huaweicloud/services/dcs/resource_huaweicloud_dcs_account.go

```
@@ -0,0 +1,519 @@
+
+	if err != nil {
+		return diag.Errorf("error creating DCS client: %s", err)
+	}
+
+	instanceId := d.Get("instance_id").(string)
+	accountName := d.Get("account_name").(string)
+	createAccountPath := createAccountClient.Endpoint + createAccountHttpUrl
+	createAccountPath = strings.ReplaceAll(createAccountPath, "{project_id}", createAccountClient.ProjectID)
+	createAccountPath = strings.ReplaceAll(createAccountPath, "{instance_id}", instanceId)
+
+	createAccountOpt := golangsdk.RequestOpts{KeepResponseBody: true}
+	createAccountOpt.JSONBody = utils.RemoveNil(buildCreateAccountBodyParams(d))
+
+	retryFunc := func() (interface{}, bool, error) {
+		createAccountResp, createErr := createAccountClient.Request("POST", createAccountPath, &createAccountOpt)
+		retry, err := handleOperationError(createErr)
+		return createAccountResp, retry, err
+	}
+	r, err := common.RetryContextWithWaitForState(&common.RetryContextWithWaitForStateParam{
+		Ctx:          ctx,
+		RetryFunc:    retryFunc,
+		WaitFunc:     refreshDcsInstanceState(createAccountClient, instanceId),
+		WaitTarget:   []string{"RUNNING"},
+		Timeout:      d.Timeout(schema.TimeoutCreate),
+		DelayTimeout: 10 * time.Second,
+		PollInterval: 10 * time.Second,
+	})
+
+	if err != nil {
+		return diag.Errorf("error creating DCS account: %v", err)
+	}
+
+	_, err = utils.FlattenResponse(r.(*http.Response))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"CREATING"},
+		Target:       []string{"AVAILABLE"},
+		Refresh:      accountStatusRefreshFunc(instanceId, accountName, createAccountClient),
+		Timeout:      d.Timeout(schema.TimeoutCreate),
+		Delay:        2 * time.Second,
+		PollInterval: 2 * time.Second,
+	}
+
+	account, err := stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.Errorf("error waiting for the account (%s) to complete: %s", accountName, err)
+	}
+
+	d.SetId(utils.PathSearch("account_id", account, "").(string))
```

we should return an error is account can't be found

---

done

---

I mean update the code like this
```suggestion
        accountID := utils.PathSearch("account_id", account, nil)
        if accountID == nil {
		return diag.Errorf("the account (%s) is not found: %s", accountName, err)
	}
	d.SetId(utils.PathSearch(accountID.(string))
```

---

done